### PR TITLE
Add development documentation for podman users

### PR DESCRIPTION
**Pull request by paramite**
## Description

Adds required steps to README about running end to end tests on kind using podman.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

```Adds required steps to README about running end to end tests on kind using podman.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,14 +175,28 @@ kubectl delete --ignore-not-found customresourcedefinitions \
 #### Running *end-to-end* tests on local kind cluster:
 
 1. `kind create cluster --image=kindest/node:<latest>`. e.g `v1.23.0` version.
+
+> Note: In case you are running kind on MacOS using podman, create podman machine with at least 4 CPUs and 8GiB memory:
```
Comment (fpetkovski) [inline README.md:179]: Is the 4CPU and 8GiB memory a requirement, or just an example?

Comment (paramite) [reply README.md:179]: Using 4 CPUs on my environment end to end tests started passing. If it is a requirement or just an accidental value I don't know. But having just one or two CPUs podman machine makes end to end tests timeout because operator fails to deploy complaining on lack of CPU resources. The value for 8GiB memory I got from some googled documentation  (recommended value, minimal was 4GB), I don't have at hand now, but can google it again in case it's required.

Comment (fpetkovski) [reply README.md:179]: Ok, thanks for the detailed explanation! Maybe we can just reword it to say that 4 CPUs and 8GiB is recommended for podman machine in general, so that people don't think this is what the operator needs :)

Comment (paramite) [reply README.md:179]: Fair enough, will reword it then.

**Review by fpetkovski — APPROVED** [review]

**Comment by paramite** [comment]
Does anybody know what's wrong with the code indentation? I tried 4 spaces, 2 spaces and no space and still the doc check complains (when there was 4 it suggested 2, when there was 2 it suggested none and when there was none it suggested 2 :D ... I give up)

File: README.md [hunk 2]
```diff
@@ -175,14 +175,32 @@ kubectl delete --ignore-not-found customresourcedefinitions \
...
+>
+>  `podman machine init --cpus=4 --memory=8192 --rootful --now`
+>
+
 2. `kubectl cluster-info --context kind-kind`. kind version >= 0.6.x
 3. `make image` - build Prometheus Operator docker image locally.
+
+> Note: In case you are running kind using podman, the step 3 won't work for you. You will need to switch command in Makefile:
+>
+>  `sed -i 's/docker build/podman build/g' Makefile`
```
Comment (simonpasquier) [inline README.md:190]: this could also be "fixed" in the Makefile by adding a variable with a default value of `docker`:

```
DOCKER_CLI ?= docker
...
.hack-operator-image: Dockerfile operator
    $(DOCKER_CLI) build
```

Then you can run `DOCKER_CLI=podman make image`

**Review by simonpasquier — COMMENTED** [review]
@paramite have you tried running `make docs`?

File: README.md (hunk 2 above)
Comment (paramite) [reply README.md]: I did not. Thanks!

**Comment by simonpasquier** [comment]
Thanks! Feel free to open another PR for a more friendly `Makefile` as I've suggested in https://github.com/prometheus-operator/prometheus-operator/pull/4754#discussion_r863849812.

**Comment by paramite** [comment]
Somehow I missed that comment. That's a good point, will do.
